crates/kubelet/src/container/state.rs

```diff
@@ -20,26 +23,41 @@ pub async fn run_to_completion<S: ResourceState<Manifest = Container, Status = S
     initial_state: impl State<S>,
     shared: SharedState<S::SharedState>,
     mut container_state: S,
-    pod: SharedState<Pod>,
+    pod: Manifest<Pod>,
     container_name: ContainerKey,
 ) -> anyhow::Result<()> {
-    let (pod_name, api) = {
-        let initial_pod = pod.read().await.clone();
-        let namespace = initial_pod.namespace().to_string();
-        let name = initial_pod.name().to_string();
-        let api: Api<KubePod> = Api::namespaced(client.clone(), &namespace);
-        (name, api)
-    };
+    let initial_pod = pod.latest();
+    let namespace = initial_pod.namespace().to_string();
+    let pod_name = initial_pod.name().to_string();
+    let api: Api<KubePod> = Api::namespaced(client.clone(), &namespace);
 
     let mut state: Box<dyn State<S>> = Box::new(initial_state);
 
+    // Forward pod updates as container updates.
+    let initial_container = initial_pod.find_container(&container_name).unwrap();
+    let (container_tx, container_rx) = Manifest::new(initial_container);
+    let mut task_pod = pod.clone();
+    let task_container_name = container_name.clone();
+    tokio::spawn(async move {
+        while let Some(latest_pod) = task_pod.next().await {
```

Comment:
Here is the only example right now of using the `Stream` API. I use it to detect pod updates and forward changes to individual containers to the `Manifest<Container>` used in the container state machine.